src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,62 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final int FIGURE_COUNT = 5;
+
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        return switch (figureNumber) {
+
+            case 0 -> new Circle(random.nextDouble() * 10 + 1,
+                    AbstractFigure.getRandomColor()) {
+                @Override
+                public Color getColor() {
+                    return null;
+                }
+            };
+            case 1 -> new Rectangle(random.nextDouble() * 10 + 1,
+                    random.nextDouble() * 10 + 1,
+                    AbstractFigure.getRandomColor()) {
+                @Override
```

Comment:
create a private method to get random color, and reuse it for each figure
you can also extract code of figure creation to dedicated private methods, if that will make this class easier to read